File: app/src/main/java/com/mathematicalfunk/appterminator/application/ProcessViewHolderAdapter.java
package com.mathematicalfunk.appterminator.application;

import android.app.ActivityManager;
import android.content.pm.ApplicationInfo;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.mathematicalfunk.appterminator.R;

import java.util.List;


public class ProcessViewHolderAdapter extends RecyclerView.Adapter<ProcessViewHolder> {
    private List<ApplicationInfo> applicationInfoList;
    private ActivityManager activityManager;

    public ProcessViewHolderAdapter(ActivityManager activityManager) {
        this.activityManager = activityManager;
    }

    @Override
    public ProcessViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.process_list_item, parent, false);
        return new ProcessViewHolder(view, activityManager);
    }

    @Override
    public void onBindViewHolder(ProcessViewHolder holder, int position) {
        ApplicationInfo applicationInfo = applicationInfoList.get(position);
        holder.bindView(applicationInfo);
    }

    @Override
    public int getItemCount() {
        return applicationInfoList.size();
    }

    public void addProcesses(final List<ApplicationInfo> applicationInfoList) {
        this.applicationInfoList = applicationInfoList;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/emmettwilson/dev/appterminator/application/ProcessViewHolderAdapter.kt
package com.emmettwilson.dev.appterminator.application

import android.app.ActivityManager
import android.content.pm.ApplicationInfo
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.emmettwilson.appterminator.R

class ProcessViewHolderAdapter(private val activityManager: ActivityManager) : RecyclerView.Adapter<ProcessViewHolder>() {
    private var applicationInfoList: List<ApplicationInfo>? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProcessViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.process_list_item, parent, false)
        return ProcessViewHolder(view, activityManager)
    }

    override fun onBindViewHolder(holder: ProcessViewHolder, position: Int) {
        val applicationInfo = applicationInfoList!![position]
        holder.bindView(applicationInfo)
    }

    override fun getItemCount(): Int {
        return applicationInfoList!!.size
    }

    fun addProcesses(applicationInfoList: List<ApplicationInfo>?) {
        this.applicationInfoList = applicationInfoList
        notifyDataSetChanged()
    }
}
